{"url": "https://api.github.com/repos/rust-lang/rust/issues/10626", "repository_url": "https://api.github.com/repos/rust-lang/rust", "labels_url": "https://api.github.com/repos/rust-lang/rust/issues/10626/labels{/name}", "comments_url": "https://api.github.com/repos/rust-lang/rust/issues/10626/comments", "events_url": "https://api.github.com/repos/rust-lang/rust/issues/10626/events", "html_url": "https://github.com/rust-lang/rust/issues/10626", "id": 23198823, "node_id": "MDU6SXNzdWUyMzE5ODgyMw==", "number": 10626, "title": "logging macros and writeln!() crash if stderr is redirected to /dev/null", "user": {"login": "dwrensha", "id": 495768, "node_id": "MDQ6VXNlcjQ5NTc2OA==", "avatar_url": "https://avatars.githubusercontent.com/u/495768?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dwrensha", "html_url": "https://github.com/dwrensha", "followers_url": "https://api.github.com/users/dwrensha/followers", "following_url": "https://api.github.com/users/dwrensha/following{/other_user}", "gists_url": "https://api.github.com/users/dwrensha/gists{/gist_id}", "starred_url": "https://api.github.com/users/dwrensha/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dwrensha/subscriptions", "organizations_url": "https://api.github.com/users/dwrensha/orgs", "repos_url": "https://api.github.com/users/dwrensha/repos", "events_url": "https://api.github.com/users/dwrensha/events{/privacy}", "received_events_url": "https://api.github.com/users/dwrensha/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2013-11-24T02:41:32Z", "updated_at": "2013-11-25T07:51:47Z", "closed_at": "2013-11-25T07:51:47Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "The following occurs for me on Mac OSX 10.9.\n\n```\n// child.rs\nfn main() {\n    for _ in range::<uint>(0, 1000) {\n        error!(\"hello?\");\n    }\n}\n```\n\n```\n// parent.rs\nfn main () {\n    use std::io::process;\n\n    let config = process::ProcessConfig {\n        program : \"./child\",\n        args : [],\n        env : None,\n        cwd : None,\n        io : []\n    };\n\n    match process::Process::new(config) {\n        Some(ref mut p) => println!(\"{}\", p.wait()),\n        None => println(\"failed to start child\")\n    }\n}\n```\n\nWhen I run `./parent`, I expect the output to be \"exit code: 0\". Instead, I see an error signal, and not always the same one:\n\n```\n$ ./parent\nsignal: 6\n$ ./parent\nsignal: 4\n$ ./parent\nsignal: 6\n$ ./parent\nsignal: 6\n$ ./parent\nsignal: 10\n```\n", "closed_by": {"login": "alexcrichton", "id": 64996, "node_id": "MDQ6VXNlcjY0OTk2", "avatar_url": "https://avatars.githubusercontent.com/u/64996?v=4", "gravatar_id": "", "url": "https://api.github.com/users/alexcrichton", "html_url": "https://github.com/alexcrichton", "followers_url": "https://api.github.com/users/alexcrichton/followers", "following_url": "https://api.github.com/users/alexcrichton/following{/other_user}", "gists_url": "https://api.github.com/users/alexcrichton/gists{/gist_id}", "starred_url": "https://api.github.com/users/alexcrichton/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/alexcrichton/subscriptions", "organizations_url": "https://api.github.com/users/alexcrichton/orgs", "repos_url": "https://api.github.com/users/alexcrichton/repos", "events_url": "https://api.github.com/users/alexcrichton/events{/privacy}", "received_events_url": "https://api.github.com/users/alexcrichton/received_events", "type": "User", "site_admin": false}, "reactions": {"url": "https://api.github.com/repos/rust-lang/rust/issues/10626/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/rust-lang/rust/issues/10626/timeline", "performed_via_github_app": null, "state_reason": "completed"}